sdk/identity/azure-identity/tests/test_known_authority_hosts.py

```diff
@@ -0,0 +1,16 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from azure.identity._internal.msal_credentials import MsalCredential
+
+
+def test_set_known_authority_host():
+    known_authority_hosts = ["private.cloud"]
+    credential = MsalCredential(
+        client_id="CLIENT_ID",
+        known_authority_hosts=known_authority_hosts,
+    )
+    app = credential._get_app()
+    expected = frozenset(known_authority_hosts)
+    assert app._known_authority_hosts == expected
```

Comment:
I'm fine to keep this test case as-is. Just want to call out that it relies on an upstream internal attribute, so it may potentially break in the future. You can consider porting [this upstream test case](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/492/files#diff-1982837bc801e709175e25839e1046a445df689d7a2e6309ccb6856e350dee7eR187-R191) instead. That is the public behavior that MSAL will promise for backward compatibility.

Reply:
Looks like the test you linked was to call set_known_authority_hosts twice?

Reply:
Indeed. It mimics the calling pattern in Azure Identity that multiple MsalCredential(..., known_authority_hosts=[...]) could be used.